code/modules/mob/living/carbon/xenomorph/castes/Spitter.dm

```diff
@@ -74,11 +74,10 @@
 		if (H.stat == DEAD)
 			return
 
-	for (var/atom/dotA in dot_cooldown_atoms)
-		if (dotA == A)
-			return
+		if(WEAKREF(A) in dot_cooldown_atoms)
+		return
```

Comment:
At a quick glance I assume this return is meant to be indented so as it be in the if()